### PR TITLE
Sync visual shader's distance fade using circular fade

### LINKED ISSUE #86367: VisualShader DistanceFade inconsistent with StandardMaterial3D
### Tested versions

Godot v4.2.stable.mono
Godot v4.3.dev.mono (335ff0f1a)

### System information

Godot v4.3.dev.mono (335ff0f1a) - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3070 Laptop GPU (NVIDIA; 31.0.15.3742) - 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz (16 Threads)

### Issue description

I have noticed that some inconsistencies exist between what the VisualShader functions produce and what StandardMaterial3D generate/use. In most cases the VisualShader version is wrong or inherently worse.

The one I want to flag here to keep this a single unit is VisualShader DistanceFunction produces this (bad):
float n_out9p0 = clamp(smoothstep(n_in9p0, n_in9p1,**-VERTEX.z**),0.0,1.0);
image

3dMaterial + converted shader produces this (good):
ALPHA *= clamp(smoothstep(distance_fade_min, distance_fade_max, **length(VERTEX)**), 0.0, 1.0);
image

Here they are side by side (left/blue=VisualShader right/yellow=3dMaterial:
image

### Steps to reproduce

Create StandardMaterial3D and use 
Create ShaderMaterial and use DistanceFunction

### Minimal reproduction project (MRP)

[DF_MRP.zip](https://github.com/godotengine/godot/files/13733299/DF_MRP.zip)

### PR DESCRIPTION
**Pull request by jsjtxietian**
Fixes https://github.com/godotengine/godot/issues/86367

### CODE REVIEW

**Review by paddy-exe — APPROVED** [review]
LGTM and works as expected now:
image

File: scene/resources/visual_shader_nodes.cpp [hunk 1]
```diff
@@ -7825,7 +7825,7 @@ bool VisualShaderNodeDistanceFade::has_output_port_preview(int p_port) const {
 
 String VisualShaderNodeDistanceFade::generate_code(Shader::Mode p_mode, VisualShader::Type p_type, int p_id, const String *p_input_vars, const String *p_output_vars, bool p_for_preview) const {
 	String code;
-	code += vformat("	%s = clamp(smoothstep(%s, %s,-VERTEX.z),0.0,1.0);\n", p_output_vars[0], p_input_vars[0], p_input_vars[1]);
+	code += vformat("	%s = clamp(smoothstep(%s, %s,length(VERTEX)),0.0,1.0);\n", p_output_vars[0], p_input_vars[0], p_input_vars[1]);
```
Comment (AThousandShips) [inline scene/resources/visual_shader_nodes.cpp:7828]:
```suggestion
	code += vformat("	%s = clamp(smoothstep(%s, %s, length(VERTEX)), 0.0, 1.0);\n", p_output_vars[0], p_input_vars[0], p_input_vars[1]);
```
Let's just clean it up while we're here

Comment (jsjtxietian) [reply scene/resources/visual_shader_nodes.cpp:7828]: Yes, thanks!

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**